--- data/org.cinnamon.gschema.xml.in	2019-01-08 11:38:41.000000000 -0500
+++ data/org.cinnamon.gschema.xml.in.new	2019-02-27 22:56:36.257133578 -0500
@@ -657,7 +657,7 @@
   <schema id="org.cinnamon.theme" path="/org/cinnamon/theme/"
         gettext-domain="@GETTEXT_PACKAGE@">
     <key name="name" type="s">
-      <default>""</default>
+      <default>"Mint-Y"</default>
       <_summary>Theme name</_summary>
       <_description>
        The name of the theme
